rclcpp/src/rclcpp/executors/static_single_threaded_executor.cpp

```diff
@@ -39,13 +39,18 @@ StaticSingleThreadedExecutor::spin()
   // except we need to keep the wait result to reproduce the StaticSingleThreadedExecutor
   // behavior.
   while (rclcpp::ok(this->context_) && spinning.load()) {
-    this->spin_once_impl(std::chrono::nanoseconds(-1));
+    return this->spin_some_impl(std::chrono::nanoseconds(0), true);
```

Comment:
Why `return` ?
This is `spin`, which should be a blocking method that returns void

Reply:
because I should not code before I had my second coffee ;-)

Reply:
Updated the code...